### PR TITLE
Do not allow concurrent persists by default

### PR DESCRIPTION
**Pull request by freekmurze**

### CODE REVIEW

File: src/AggregateRoot.php [hunk 1]
```diff
@@ -6,19 +6,25 @@
...
 use ReflectionClass;
 use ReflectionProperty;
+use Spatie\EventSourcing\Exceptions\CouldNotPersistAggregate;
 use Spatie\EventSourcing\Snapshots\Snapshot;
 use Spatie\EventSourcing\Snapshots\SnapshotRepository;
 
 abstract class AggregateRoot
 {
-    private string $aggregateUuid;
+    private string $aggregateUuid = '';
```
Comment (riasvdv) [inline src/AggregateRoot.php]: Rename aggregateProperties without "aggregate", since we are in an aggregate class

File: src/AggregateRoot.php [hunk 2]
```diff
@@ -31,7 +37,8 @@ public static function retrieve(string $uuid): AggregateRoot
     }
 
     /**
-     * @param  ShouldBeStored  $domainEvent
+     * @param ShouldBeStored $domainEvent
+     *
```
Comment (riasvdv) [inline src/AggregateRoot.php]: Do we need this docblock?

File: src/StoredEventRepository.php [hunk 3]
```diff
@@ -17,4 +17,6 @@ public function persist(ShouldBeStored $event, string $uuid = null, int $aggrega
     public function persistMany(array $events, string $uuid = null, int $aggregateVersion = null): LazyCollection;
 
     public function update(StoredEvent $storedEvent): StoredEvent;
+
+    public function getLatestVersion(string $aggregateUuid): int;
```
Comment (riasvdv) [inline src/StoredEventRepository.php]: Maybe we should rename this to `->getLatestAggregateVersion` so it's clear to someone implementing the interface that they need to return the aggregate's version

File: src/AggregateRoot.php (hunk 1 above)
Comment (freekmurze) [reply src/AggregateRoot.php]: Rename `aggregateUuid` to `uuid`.

File: src/AggregateRoot.php (hunk 2 above)
Comment (freekmurze) [reply src/AggregateRoot.php]: Removed!

File: src/StoredEventRepository.php (hunk 3 above)
Comment (freekmurze) [reply src/StoredEventRepository.php]: Good idea. Done!

**Pull request merged**